spec/eof.md

```diff
@@ -327,28 +327,48 @@ The following instructions are introduced in EOF code:
+- *Forward jump* refers to any of `RJUMP`/`RJUMPI`/`RJUMPV` instruction with relative offset greater than or equal to 0. *Backwards jump* refers to any of `RJUMP`/`RJUMPI`/`RJUMPV` instruction with relative offset less than 0, including jumps to the same jump instruction (e.g. `RJUMP(-3)`)
+- Terminating instructions: 
+  - ending function execution: `RETF`, `JUMPF`,
+  - ending whole EVM execution: `STOP`, `RETURN`, `RETURNCONTRACT`, `REVERT`, `INVALID`.
 - For each instruction in the code the operand stack height bounds are recorded as `stack_height_min` and `stack_height_max`. Instructions are scanned in a single linear pass over the code.
-- During scanning:
-  - first instruction has `stack_height_min = stack_height_max = types[current_section_index].inputs`;
-  - for each instruction reached from a forwards jump or sequential flow from previous instruction, update the target bounds so that they contain source bounds, i.e. `target_stack_min = min(target_stack_min, source_stack_min + change)` and `target_stack_max = max(target_stack_max, source_stack_max + change)`. `change` is the stack height change of the source instruction OPCODE or `outputs - inputs` in case of `CALLF`;
-  - for each instruction reached from a backwards jump, check if target bounds are same as source bounds, i.e. `target_stack_min == source_stack_min + change` and `target_stack_max == source_stack_max + change`.
-- No instruction may access more operand stack items than `stack_height_min`
-- Terminating instructions: `STOP`, `INVALID`, `RETURN`, `REVERT`, `RETURNCONTRACT`, `RETF`, `JUMPF`.
-- During `CALLF`, the following must hold: `stack_height_min >= types[target_section_index].inputs`
-- During `CALLF` and `JUMPF`, the following must hold: `stack_height_max + types[target_section_index].max_stack_height - types[target_section_index].inputs <= 1024`
-- Stack validation of `JUMPF` depends on "non-returning" status of target section
-    - `JUMPF` into returning section (can be only from returning section): `stack_height_min == stack_height_max == type[current_section_index].outputs + type[target_section_index].inputs - type[target_section_index].outputs`
-    - `JUMPF` into non-returning section: `stack_height_min >= types[target_section_index].inputs`
-- During `RETF`, the following must hold: `stack_height_max == stack_height_min == types[current_code_index].outputs`
-- During terminating instructions `STOP`, `INVALID`, `RETURN`, `REVERT`, `RETURNCONTRACT` operand stack may contain extra items below ones required by the instruction
-- the last instruction may be a terminating instruction or `RJUMP`
-- no instruction may be unreachable
+- first instruction has `stack_height_min = stack_height_max = types[current_section_index].inputs`;
+
+- During scanning, for each instruction:
```

Comment:
In readable version formatting is broken. It starts from point 3...

Reply:
For some reason it required 3-spaces indentation as in EIP, but hopefully fixed now.